### ISSUE TITLE
Using BN+WS in ResNet50 similar or slightly worse than using BN only in ResNet50 for ImageNet classification

### ISSUE THREAD

**Issue by niam06** [joe-siyuan-qiao/WeightStandardization#13]
Thanks for your nice work on weight standardization. Weight standardization seemed to work better on CIFAR and Jester dataset, however on ImageNet, results seem to be slightly worse with BN+WS than with BN alone (around 76.0% with BN alone vs around 75.9% with BN+WS). I used this script https://github.com/pytorch/examples/tree/master/imagenet and used a modified ResNet50 as you recommended in here. Is the effectiveness of WS considerable only when compared with GN? Please let me know of your thoughts and experience on this.

**Comment by joe-siyuan-qiao** [comment]
Hi, we have done experiments on ImageNet and found that WS+BN is consistently better than BN alone. You can find them in Table 1 of here https://arxiv.org/pdf/1903.10520.pdf. The experiments were done using the code here https://github.com/joe-siyuan-qiao/pytorch-classification. Thanks.

**Comment by niam06** [comment]
Thanks a lot for your quick response! I will look into it.

**Comment by niam06** [comment]
With the code you linked, results of WS is better - thanks for helping out! Great work! 

One other question - have you shared (or do you intend to share) the code to plot the rightmost figure of Figure 4? It would be great to see the training dynamics in terms of reduction percentages of Lipschitz constant.

**Comment by joe-siyuan-qiao** [comment]
No, we haven't shared that code. We used the hook function in PyTorch to record the gradients of the intermediate variables, based on which we computed the reduction percentages.
